plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -284,6 +287,217 @@ public void testSelectivelyOptimizingLeavesEqualityDeletes()
         assertThat(loadTable(tableName).currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("1");
     }
 
+    @Test
+    public void testMultipleEqualityDeletes()
+            throws Exception
+    {
+        String tableName = "test_equality_delets_different_schemas_" + randomNameSuffix();
+        assertUpdate("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.tiny.nation", 25);
+        Table icebergTable = loadTable(tableName);
+        assertThat(icebergTable.currentSnapshot().summary().get("total-equality-deletes")).isEqualTo("0");
+        Path metadataDir = new Path(metastoreDir.toURI());
+        TrinoFileSystem fs = HDFS_FILE_SYSTEM_FACTORY.create(SESSION);
+
+        List<String> deleteFileColumns = ImmutableList.of("regionkey");
+        Schema deleteRowSchema = icebergTable.schema().select(deleteFileColumns);
+        List<Integer> equalityFieldIds = deleteFileColumns.stream()
+                .map(name -> deleteRowSchema.findField(name).fieldId())
+                .collect(toImmutableList());
+        for (int i = 1; i < 3; i++) {
+            String deleteFile = "delete_file_" + UUID.randomUUID();
+            Parquet.DeleteWriteBuilder writerBuilder = Parquet.writeDeletes(new ForwardingFileIo(fs).newOutputFile(new Path(metadataDir, deleteFile).toString()))
+                    .forTable(icebergTable)
+                    .rowSchema(deleteRowSchema)
+                    .createWriterFunc(GenericParquetWriter::buildWriter)
+                    .equalityFieldIds(equalityFieldIds)
+                    .overwrite();
+            EqualityDeleteWriter<Record> writer = writerBuilder.buildEqualityWriter();
+
+            Record dataDelete = GenericRecord.create(deleteRowSchema);
+            try (Closeable ignored = writer) {
+                writer.write(dataDelete.copy(ImmutableMap.of("regionkey", 1L)));
+            }
+            icebergTable.newRowDelta().addDeletes(writer.toDeleteFile()).commit();
+        }
+
+        assertQuery("SELECT * FROM " + tableName, "SELECT * FROM nation WHERE  regionkey != 1L");
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testEqualityDeletesWithDifferentSchemas()
```

Comment:
optional:

Given that `org.apache.iceberg.types.TypeUtil#select(org.apache.iceberg.types.Types.StructType, java.util.Set<java.lang.Integer>)` produces equivalent results when providing the IDs corresponding to the column name combinations `"regionkey", "name"` or `"name", "regionkey"` we could name the test

`testEqualityDeletesWithEquivalentSchemas`